### PR TITLE
Implement new vat check Netherlands

### PR DESCRIPTION
**Pull request by rconjour**
Due to legal changes, some businesses in The Netherlands got a new vat number, these changes are backported from Odoo 12 and modified.

### CODE REVIEW

File: addons/base_vat/base_vat.py [hunk 1]
```diff
@@ -227,11 +227,15 @@ def check_vat_ie(self, vat):
 
     # Mexican VAT verification, contributed by Vauxoo
     # and Panos Christeas <p_christ@hol.gr>
-    __check_vat_mx_re = re.compile(r"(?P<primeras>[A-Za-z\xd1\xf1&]{3,4})" \
-                                    r"[ \-_]?" \
-                                    r"(?P<ano>[0-9]{2})(?P<mes>[01][0-9])(?P<dia>[0-3][0-9])" \
-                                    r"[ \-_]?" \
-                                    r"(?P<code>[A-Za-z0-9&\xd1\xf1]{3})$")
+    __check_vat_mx_re = re.compile(br"(?P<primeras>[A-Za-z\xd1\xf1&]{3,4})" \
```
Comment (StefanRijnhart) [inline addons/base_vat/base_vat.py]: Is this a whitespace change? Can you revert it, as it seems unrelated to this PR.

Comment (rconjour) [reply addons/base_vat/base_vat.py]: It is a change of `r` to `br`, shall i revert this one?

File: addons/base_vat/base_vat.py [hunk 2]
```diff
@@ -359,4 +363,51 @@ def check_vat_tr(self, vat):
 
         return False
 
-# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
+    def check_vat_nl(self, vat):
+        """
+        Temporary Netherlands VAT validation to support the new format
```
Comment (StefanRijnhart) [inline addons/base_vat/base_vat.py]: Is this the commit that you cherry picked? I'm surprised that the comment is different: https://github.com/odoo/odoo/commit/3ddaf382cb353829bcd226f114abcf4c8e0f0157#diff-5818ba1e95e0716e509401478e1aacc8R387-R392

File: addons/base_vat/base_vat.py (hunk 1 above)
Comment (StefanRijnhart) [reply addons/base_vat/base_vat.py]: Yes please!

File: addons/base_vat/base_vat.py [hunk 3]
```diff
@@ -359,4 +363,57 @@ def check_vat_tr(self, vat):
...
+        TODO: remove when fixed upstream
+        """
+
+        try:
+            from stdnum.util import clean
+            from stdnum.nl.bsn import checksum
+        except ImportError:
+            return True
+
+        vat = "NL" + vat
```
Comment (StefanRijnhart) [inline addons/base_vat/base_vat.py:383]: Can you add a comment that this is needed up until Odoo 10.0 (because Odoo 11.0 and up contain this commit: https://github.com/odoo/odoo/commit/f643bca6cf2e).

Comment (rconjour) [reply addons/base_vat/base_vat.py:383]: Done!

Comment (StefanRijnhart) [reply addons/base_vat/base_vat.py:383]: Thanks! The VAT validation is a bit of a mess after that commit so I've created https://github.com/odoo/odoo/pull/45095. It does not apply to Odoo 8.0

**Review by StefanRijnhart — APPROVED** [review]
Thanks! Test failures are unrelated to this PR. I ran the tests successfully locally.

**Comment by NL66278** [comment]
Is this still needed? Just updating python-stdnum to 1.13 will fix the problem. Also for 8.0.

**Comment by sergiocorato** [comment]
> Is this still needed? Just updating python-stdnum to 1.13 will fix the problem. Also for 8.0.

Odoo doesn't use this library AFAIK

**Comment by NL66278** [comment]
@sergiocorato Odoo does use stdnum (and vatnumber that depends on stdnum) to do vat validation:
https://github.com/odoo/odoo/blob/13.0/addons/base_vat/models/res_partner.py#L10

In 8.0 it was only indirectly through vatnumber:
https://github.com/odoo/odoo/blob/8.0/addons/base_vat/base_vat.py#L28

**Comment by sergiocorato** [comment]
> @sergiocorato Odoo does use stdnum (and vatnumber that depends on stdnum) to do vat validation:
> https://github.com/odoo/odoo/blob/13.0/addons/base_vat/models/res_partner.py#L10
> 
> In 8.0 it was only indirectly through vatnumber:
> https://github.com/odoo/odoo/blob/8.0/addons/base_vat/base_vat.py#L28

you're right, it wasn't installed, but installing it it solved
:+1:

**Comment by NL66278** [comment]
@sergiocorato You can close this PR then (I am not a maintainer of this repo, so can't do it myself).

**Review by sergiocorato — CHANGES_REQUESTED** [review]
Not needed as of https://github.com/OCA/OCB/pull/907#issuecomment-591310880

**Comment by sergiocorato** [comment]
> @sergiocorato You can close this PR then (I am not a maintainer of this repo, so can't do it myself).

I'm not a maintainer of this repo too, I can only put a negative review

**Review by NL66278 — CHANGES_REQUESTED** [review]
Please close this PR, as it is not needed to fix the problem described.

**Comment by NL66278** [comment]
@rconjour Please close this PR. See discussion.

**Comment by rconjour** [comment]
Why do we want to choose a different approach then the approach of Odoo? I guess we need to make another PR if we choose your approach that forces the right stdnum version in requirements.txt

**Comment by StefanRijnhart** [comment]
Thanks for informing us @NL66278 about this being in stdnum. I had only checked vatnumber and noticed it had not been updated since 2014.

**Comment by StefanRijnhart** [comment]
@rconjour Odoo has this `stable` policy so that users are guaranteed not to have to upgrade any libs or modules during the release cycle. But if you use OCB, you probably do not fear updating your libs.
